test/e2e/metallb/metallb.go

```diff
@@ -0,0 +1,111 @@
+package metallb
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	metallbv1beta1 "github.com/metallb/metallb-operator/api/v1beta1"
+	"github.com/metallb/metallb-operator/pkg/status"
+	"github.com/metallb/metallb-operator/test/consts"
+	testclient "github.com/metallb/metallb-operator/test/e2e/client"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// Timeout and Interval settings
+	Timeout  = time.Minute * 3
+	Interval = time.Second * 2
+)
+
+// Delete and check the MetalLB custom resource is deleted to avoid status leak in between tests.
+func Delete(metallb *metallbv1beta1.MetalLB) {
+	err := testclient.Client.Delete(context.Background(), metallb)
+	if errors.IsNotFound(err) { // Ignore err, could be already deleted.
+		return
+	}
+	Expect(err).ToNot(HaveOccurred())
+
+	Eventually(func() bool {
+		err := testclient.Client.Get(context.Background(), goclient.ObjectKey{Namespace: metallb.Namespace, Name: metallb.Name}, metallb)
+		return errors.IsNotFound(err)
+	}, 1*time.Minute, 5*time.Second).Should(BeTrue(), "Failed to delete MetalLB custom resource")
+
+	Eventually(func() bool {
+		_, err := testclient.Client.Deployments(metallb.Namespace).Get(context.Background(), consts.MetalLBDeploymentName, metav1.GetOptions{})
+		return errors.IsNotFound(err)
+	}, Timeout, Interval).Should(BeTrue())
+
+	Eventually(func() bool {
+		_, err := testclient.Client.DaemonSets(metallb.Namespace).Get(context.Background(), consts.MetalLBDaemonsetName, metav1.GetOptions{})
+		return errors.IsNotFound(err)
+	}, Timeout, Interval).Should(BeTrue())
+
+	Eventually(func() bool {
+		pods, _ := testclient.Client.Pods(metallb.Namespace).List(context.Background(), metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("component=%s", consts.MetalLBDeploymentName)})
+		return len(pods.Items) == 0
+	}, Timeout, Interval).Should(BeTrue())
+
+	Eventually(func() bool {
+		pods, _ := testclient.Client.Pods(metallb.Namespace).List(context.Background(), metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("component=%s", consts.MetalLBDaemonsetName)})
+		return len(pods.Items) == 0
+	}, Timeout, Interval).Should(BeTrue())
+}
+
+func Get(operatorNamespace string, useMetallbResourcesFromFile bool) (*metallbv1beta1.MetalLB, error) {
```

Comment:
Can you add a description here of what the Get is retrieving?

Reply:
This is a valid, good question. Please note, though, that this is a release-4.9 downstream backport. It would be best to add a description upstream first and cascade it downstream afterward, which can be done aside from this PR.

Reply:
Ok great!